<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Geoff Lawler</author>
 <class>messageStreamPlaybackForm</class>
 <widget class="QWidget" name="messageStreamPlaybackForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>720</width>
    <height>120</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="rewindToStartButton">
       <property name="toolTip">
        <string>Rewind to start of data</string>
       </property>
       <property name="statusTip">
        <string>Rewind to start of data</string>
       </property>
       <property name="whatsThis">
        <string>Rewind to start of data</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../resources/QMessageStreamPlaybackWidget.qrc">
         <normaloff>:/buttonImages/gtk-media-previous-ltr.png</normaloff>:/buttonImages/gtk-media-previous-ltr.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="rewindButton">
       <property name="toolTip">
        <string>View stream in reverse direction at current rate  (or click again to double current rate)</string>
       </property>
       <property name="statusTip">
        <string>View stream in reverse direction at current rate  (or click again to double current rate)</string>
       </property>
       <property name="whatsThis">
        <string>Go backward in time at current rate</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../resources/QMessageStreamPlaybackWidget.qrc">
         <normaloff>:/buttonImages/gtk-media-rewind-ltr.png</normaloff>:/buttonImages/gtk-media-rewind-ltr.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="playButton">
       <property name="toolTip">
        <string>Play in forward direction at normal rate</string>
       </property>
       <property name="statusTip">
        <string>Play in forward direction at normal rate</string>
       </property>
       <property name="whatsThis">
        <string>Play in forward direction at normal rate</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../resources/QMessageStreamPlaybackWidget.qrc">
         <normaloff>:/buttonImages/gtk-media-play-ltr.png</normaloff>:/buttonImages/gtk-media-play-ltr.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pauseButton">
       <property name="toolTip">
        <string>Pause/Unpause playback</string>
       </property>
       <property name="statusTip">
        <string>Pause/Unpause playback</string>
       </property>
       <property name="whatsThis">
        <string>Pause/Unpause playback</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../resources/QMessageStreamPlaybackWidget.qrc">
         <normaloff>:/buttonImages/gtk-media-pause.png</normaloff>:/buttonImages/gtk-media-pause.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="forwardButton">
       <property name="toolTip">
        <string>View stream in forward direction at current rate (or click again to double current rate)</string>
       </property>
       <property name="statusTip">
        <string>View stream in forward direction at current rate (or click again to double current rate)</string>
       </property>
       <property name="whatsThis">
        <string>Go forward in time at current rate</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../resources/QMessageStreamPlaybackWidget.qrc">
         <normaloff>:/buttonImages/gtk-media-forward-ltr.png</normaloff>:/buttonImages/gtk-media-forward-ltr.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="forwardToEndButton">
       <property name="toolTip">
        <string>Jump to end of playback data or the live stream</string>
       </property>
       <property name="statusTip">
        <string>Jump to end of playback data or the live stream</string>
       </property>
       <property name="whatsThis">
        <string>Jump to end of playback data or the live stream</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../resources/QMessageStreamPlaybackWidget.qrc">
         <normaloff>:/buttonImages/gtk-media-next-ltr.png</normaloff>:/buttonImages/gtk-media-next-ltr.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="streamRateSpinBox">
       <property name="toolTip">
        <string>Set playback rate</string>
       </property>
       <property name="statusTip">
        <string>Set playback rate</string>
       </property>
       <property name="whatsThis">
        <string>Set playback rate</string>
       </property>
       <property name="accelerated">
        <bool>true</bool>
       </property>
       <property name="correctionMode">
        <enum>QAbstractSpinBox::CorrectToNearestValue</enum>
       </property>
       <property name="keyboardTracking">
        <bool>false</bool>
       </property>
       <property name="minimum">
        <double>-100.000000000000000</double>
       </property>
       <property name="maximum">
        <double>100.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.500000000000000</double>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="streamButton">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="minimumSize">
        <size>
         <width>148</width>
         <height>27</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Modify Stream Connection Properties</string>
       </property>
       <property name="statusTip">
        <string>Modify Stream Connection Properties</string>
       </property>
       <property name="whatsThis">
        <string>Modify Stream Connection Properties</string>
       </property>
       <property name="text">
        <string>Stream &amp;Configuration</string>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <item>
      <widget class="QSlider" name="playbackSlider">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="toolTip">
        <string>Drag and drop to desired playback location</string>
       </property>
       <property name="statusTip">
        <string>Drag and drop to desired playback location</string>
       </property>
       <property name="whatsThis">
        <string>Drag and drop to desired playback location</string>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="invertedAppearance">
        <bool>false</bool>
       </property>
       <property name="invertedControls">
        <bool>false</bool>
       </property>
       <property name="tickPosition">
        <enum>QSlider::NoTicks</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="playbackTimeLabel">
       <property name="text">
        <string>Current Time</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>of</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="maxTimeLabel">
       <property name="text">
        <string>Max Time</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="2" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="statusLabel">
       <property name="text">
        <string>Latest Status String</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="descriptionLabel">
       <property name="toolTip">
        <string>Connected Message Stream Description</string>
       </property>
       <property name="statusTip">
        <string>Connected Message Stream Description</string>
       </property>
       <property name="whatsThis">
        <string>Connected Message Stream Description</string>
       </property>
       <property name="text">
        <string>Message Stream Description Name</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../resources/QMessageStreamPlaybackWidget.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>streamButton</sender>
   <signal>pressed()</signal>
   <receiver>messageStreamPlaybackForm</receiver>
   <slot>chooseMessageStream()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>25</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>59</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <designerdata>
  <property name="gridDeltaX">
   <number>10</number>
  </property>
  <property name="gridDeltaY">
   <number>10</number>
  </property>
  <property name="gridSnapX">
   <bool>true</bool>
  </property>
  <property name="gridSnapY">
   <bool>true</bool>
  </property>
  <property name="gridVisible">
   <bool>true</bool>
  </property>
 </designerdata>
 <slots>
  <signal>playbackPaused()</signal>
  <signal>playbackNormal()</signal>
  <signal>playbackReversed()</signal>
  <signal>playbackForward()</signal>
  <signal>rewoundToStartOfPlayback()</signal>
  <signal>forwardedToEndOfPlayback()</signal>
  <signal>streamRateSet(double)</signal>
  <signal>currentTimestampUpdated(watcher::Timestamp)</signal>
  <signal>messageStreamDirectionChange()</signal>
  <slot>pausePlayback()</slot>
  <slot>normalPlayback()</slot>
  <slot>reversePlayback()</slot>
  <slot>forwardPlayback()</slot>
  <slot>rewindToStartOfPlayback()</slot>
  <slot>forwardToEndOfPlayback()</slot>
  <slot>playbackSetSpeed(double)</slot>
  <slot>setStreamPlaybackTimeFromSlider()</slot>
  <slot>messageStreamConnected(bool)</slot>
  <slot>chooseMessageStream()</slot>
 </slots>
</ui>
